README.md

```diff
@@ -91,6 +91,24 @@ As you noticed you can set expiration time for jobs globally by expiration optio
 + It is advised to set this expiration time greater than time required for completion of the job.
 + Default expiration time is 30 minutes.
 
+### Disabling recording of the initial :queued status and metadata
+By default, the client middleware records the initial metadata at job-creation.
+If this is not desired for all workers, it can be conditionally enabled
```

Comment:
Is there any use case that makes sense for this except for batched jobs? My thought here is that we might not want to expose this functionality to users at all, we might want to just disable this automatically when creating batches. We then can inform users about "what" and "why" are we doing this, but not tell them that they need to do something explicitly.

Reply:
so I am not sure how to detect batch from this layer easily

Reply:
was planning to let the worker manage this

Reply:
```ruby
module Avant
  module Sidekiq
    class Worker
      def self.record_initial_status?
        !!Thread.current[:sidekiq_batch]
      end
    end
  end
end
```

Reply:
Correct, this would be the way of detecting if the code is currently throwing jobs into a batch. I think if we make this feature opt-in then it'll be very hard to communicate to users that this timeout might happen because of this gem and that they need to create this flag. Doing this automatically is much easier for everybody.

On the negative side of things, we lose the initial status for jobs inside of batches, but I don't think we have that many batches anyway. Even if that became an issue, I would probably prefer working on bulk inserting all of the initial statuses before showing this kind of tricky flag to the user.